### DISCUSSION TITLE
建議在routing中的rules裡加入mute功能

### DISCUSSION THREAD

**Discussion by rootmelo92118** [v2fly/v2ray-core#1943] [Ideas]
有些時候部分用戶可能會因為個人原因將某些特定類型的連線進行特殊的路由行為，但有時候這些紀錄是不被需要的，以我的情況來說， 我需要對於DNS請求特別處理， 雖然v2ray可以達到目的但是Log會變得非常的雜亂，如下圖：
<img width="861" alt="截圖 2022-09-01 上午11 27 38" src="https://user-images.githubusercontent.com/32770959/187829447-54fb23b2-453b-49c4-b58d-29a698925180.png">
嚴重一點的時候，可能會幾乎佔據整個顯示匡，如下圖：
<img width="861" alt="截圖 2022-09-01 上午11 34 52" src="https://user-images.githubusercontent.com/32770959/187829537-e78e7748-c343-4aa2-a555-f360d6535eda.png">
### 期望的更動
加入`mute`變量，讓使用者決定該內容是否要出現於Log中。（未設置變量默認為`false`）
原配置：
```
"rules": [
            {
                "inboundTag": [
                    "dns-in"
                ],
                "outboundTag": "dns-out",
                "type": "field"
            },
            {
                "network": "tcp,udp",
                "outboundTag": "direct",
                "type": "field"
            }
        ]
```
新配置：
```
"rules": [
            {
                "inboundTag": [
                    "dns-in"
                ],
                "outboundTag": "dns-out",
                "mute": true,                   //匹配到本路由的連線不記錄至Log中
                "type": "field"
            },
            {
                "network": "tcp,udp",
                "outboundTag": "direct",
                "mute": false,                  //匹配到本路由的連線正常紀錄至Log中（沒有該項就默認為 "mute": false）
                "type": "field"
            }
        ]
```
### 其他內容
其實具體來說我並不知道紀錄Log的觸發模塊是在路由時還是出站時， 也奈何本人能力有限（不會golang）， 我很難從`v2ray-core`的源代碼中來驗證我的猜想， 如果他是在出站時觸發，也可以將`mute`變量新增至`outbounds`的出站配置中。

**Comment by crazypeace** [comment]
我觉得从程序设计的角度来讲, 你这个需求即不好实现, 作用也有限.
如果你这里要加mute，那其它会产生日志的地方是不是也要加mute?

对于你遇到的问题，我的建议是将日志经过过滤之后再查看，比如：
`cat /var/log/v2ray/access.log | sed '/dns-out/d' | tail -n 50`

**Reply by rootmelo92118** [reply]
還有什麼地方會產生log，能夠舉例嗎？
